ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -585,6 +585,15 @@ func (r *RayClusterReconciler) reconcilePods(ctx context.Context, instance *rayv
 	// if RayCluster is suspended, delete all pods and skip reconcile
 	if instance.Spec.Suspend != nil && *instance.Spec.Suspend {
 		clusterLabel := client.MatchingLabels{utils.RayClusterLabelKey: instance.Name}
+		allPods := corev1.PodList{}
+		if err := r.List(ctx, &allPods, client.InNamespace(instance.Namespace), clusterLabel); err != nil {
```

Comment:
I'm not super familiar with controller-runtime's client, but I'm guessing this is a list from the informer cache, which is better than always calling DeleteCollection below.

Reply:
> I'm guessing this is a list from the informer cache

Yes, in KubeRay's setup, the `List` operation reads from the informer cache.